cirq/testing/consistent_act_on.py

```diff
@@ -41,19 +53,39 @@ def state_vector_has_stabilizer(state_vector: np.ndarray,
     return np.allclose(args.target_tensor, state_vector)
 
 
-def assert_act_on_clifford_tableau_effect_matches_unitary(val: Any) -> None:
-    """Checks that act_on with CliffordTableau generates stabilizers that
-    stabilize the final state vector. Does not work with Operations or Gates
-    expecting non-qubit Qids."""
+def assert_all_implemented_act_on_effects_match_unitary(
+        val: Any,
+        assert_tableau_implemented: bool = False,
+        assert_ch_form_implemented: bool = False) -> None:
+    """Uses val's effect on final_state_vector to check act_on(val)'s behavior.
+
+    Checks that act_on with CliffordTableau or StabilizerStateCHForm behaves
+    consistently with act_on through final state vector. Does not work with
+    Operations or Gates expecting non-qubit Qids. If either of the
+    assert_*_implmented args is true, fails if the corresponding method is not
+    implemented for the test circuit.
+
+    Args:
+        val: A gate or operation that may be an input to protocols.act_on.
+        assert_tableau_implemented: asserts that protocols.act_on() works with
+          val and ActOnCliffordTableauArgs inputs.
+        assert_ch_form_implemented: asserts that protocols.act_on() works with
+          val and ActOnStabilizerStateChFormArgs inputs.
+    """
 
     # pylint: disable=unused-variable
-    __tracebackhide__ = True
```

Comment:
Why did we remove this?

Reply:
My bad, was trying to induce some tests to fail here to verify it was working properly. Reverted.